azavea/raster-foundry
# Add registration functionality

I adapted the user registration (and 'Forgot' and 'Send Activation') code from MMW to use React.js and the RF style. I also made it so different screens correspond to different URLs and the activation screen is rendered client-side.

To test:
 * Try logging in and logging out. Try making some client-side and server-side errors, which should be listed. 
 * Try signing up. This should take you to /sign-up. Try making some client-side and server-side errors, which should be listed. Try hitting back to go to /login. 
 * Go to /sign-up and put in valid info and then follow activation link in email (that is shown in console). You should be directed to /activate where you can click the button to go to / (and you should be logged in automatically).
 * Click the Send Activation Email link. That should do the same as above.
 * Click Forgot link. Try making some client-side and server-side errors, which should be listed.

Connects #52
Connects #65
